windows_service/datadog_checks/windows_service/windows_service.py

```diff
@@ -84,13 +96,48 @@ def check(self, instance):
             for service in services_unseen:
                 # if a name doesn't match anything (wrong name or no permission to access the service), report UNKNOWN
                 status = self.UNKNOWN
+                startup_type_string = self.STARTUP_TYPE_UNKNOWN
 
                 tags = ['windows_service:{}'.format(service)]
+
                 tags.extend(custom_tags)
 
+                if instance.get('windows_service_startup_type_tag', False):
+                    tags.append('windows_service_startup_type:{}'.format(startup_type_string))
+
                 if not instance.get('disable_legacy_service_tag', False):
                     self._log_deprecation('service_tag', 'windows_service')
                     tags.append('service:{}'.format(service))
 
                 self.service_check(self.SERVICE_CHECK_NAME, status, tags=tags)
                 self.log.debug('service state for %s %s', service, status)
+
+    def _get_service_startup_type(self, scm_handle, service_name):
+        """
+        Returns a tuple that describes the startup type for the service
+          - QUERY_SERVICE_CONFIG.dwStartType
+          - SERVICE_CONFIG_DELAYED_AUTO_START_INFO.fDelayedAutostart
+        """
+        hSvc = win32service.OpenService(scm_handle, service_name, win32service.SERVICE_QUERY_CONFIG)
+        service_config = win32service.QueryServiceConfig(hSvc)
+        startup_type = service_config[1]
+        if startup_type == win32service.SERVICE_AUTO_START:
+            # Query if auto start is delayed
+            is_delayed_auto = win32service.QueryServiceConfig2(
+                hSvc, win32service.SERVICE_CONFIG_DELAYED_AUTO_START_INFO
+            )
+        else:
+            is_delayed_auto = False
```

Comment:
I wonder how the return handles are closed. win32service knows how to do it? Is it wrapping them into python object. It is probably related to the comment I have left in Event subscription handle.